ngraph/core/reference/include/ngraph/runtime/reference/loop.hpp

```diff
@@ -0,0 +1,36 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and`
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <ngraph/opsets/opset5.hpp>
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            void loop(const std::shared_ptr<Function>& body,
+                      const op::util::OutputDescriptionVector& out_descs,
+                      const op::util::InputDescriptionVector& input_descs,
+                      const opset5::Loop::SpecialBodyPorts& special_ports,
+                      const HostTensorVector& out,
```

Comment:
All other references operate with raw pointers instead of HostTensors which makes them more flexible. I propose to  change signatures of reference functions here and bellow (OK to do it in separate PR)

Reply:
I agree. @mikhail-treskin Please create a new ticket in jira to not forget, and assign me

Reply:
Yeah, it is a good point. We already discussed it today with Ivan. But looks like it is require a lot of changes and we decided to do it later (we can do it later because it is not a public API).

Reply:
ticket 43220